library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -179,6 +175,24 @@ public FileDownloadInfo create(FileDownloadInfo.Reader reader) {
             }
         }, downloadsUriProvider);
 
+        unlockStaleDownloads();
+
+        updateThread = new HandlerThread("DownloadManager-UpdateThread");
+        updateThread.start();
+        updateHandler = new Handler(updateThread.getLooper(), updateCallback);
+    }
+
+    void unlockStaleDownloads() {
+        List<String> batchesToBeUnlocked = downloadsRepository.getCurrentDownloadingBatchIds();
+
+        if (batchesToBeUnlocked.isEmpty()) {
+            batchesToBeUnlocked = downloadsRepository.getCurrentSubmittedBatchIds();
```

Comment:
I don't think that is possible to have one batch with no downloading/completed but submitted ones, so in theory or is submitted or is downloading.

We would have to avoid duplicates in there so instead of a list should be a set because a downloading one will contain submitted entries almost for sure.

Will do because maybe in parallel batches there might be an edge case